lib/cli/transaction.rs

```diff
@@ -94,22 +92,28 @@ pub fn create_transaction(
     Ok(txn)
 }
 
-/// Creates a [`Transaction`] and outputs it to a file or stdout.
+/// Creates a [`Transaction`] and outputs it to a file or stdout if the `std-fs-io` feature is enabled.
 ///
 /// As a file, the `Transaction` can subsequently be signed by other parties using [`sign_transaction_file`]
 /// and then sent to the network for execution using [`send_transaction_file`].
 ///
-/// If `force` is true, and a file exists at `transaction_params.output_path`, it will be overwritten.  If `force`
-/// is false and a file exists at `transaction_params.output_path`, [`Error::FileAlreadyExists`] is returned
-/// and the file will not be written.
+/// If the `std-fs-io` feature is NOT enabled, `maybe_output_path` and `force` are ignored.
+/// Otherwise, `maybe_output_path` specifies the output file path, or if empty, will print it to
+/// `stdout`.  If `force` is true, and a file exists at `maybe_output_path`, it will be
+/// overwritten.  If `force` is false and a file exists at `maybe_output_path`,
+/// [`Error::FileAlreadyExists`] is returned and the file will not be written.
 pub fn make_transaction(
     builder_params: TransactionBuilderParams,
     transaction_params: TransactionStrParams<'_>,
-    force: bool,
-) -> Result<(), CliError> {
-    let output = parse::output_kind(transaction_params.output_path, force);
-    let transaction = create_transaction(builder_params, transaction_params, true)?;
-    crate::output_transaction(output, &transaction).map_err(CliError::from)
+    #[allow(unused_variables)] force: bool,
+) -> Result<TransactionV1, CliError> {
+    let transaction = create_transaction(builder_params, transaction_params.clone(), true)?;
+    #[cfg(feature = "std-fs-io")]
+    {
+        let output = parse::output_kind(transaction_params.output_path, force);
+        let _ = crate::output_transaction(output, &transaction).map_err(CliError::from);
```

Comment:
```suggestion
        crate::output_transaction(output, &transaction).map_err(CliError::from)?;
```
we shouldn't ignore the error here

Reply:
Thanks should be fixed in 

https://github.com/casper-ecosystem/casper-client-rs/pull/153/commits/b4a8f436f6d31c6cc38a1ad93843d9523279dd2c